aip/general/0193.md

```diff
@@ -75,224 +84,354 @@ following:
-- It affords clients the ability to find an error message consistently
-  in one location, `LocalizedMessaage.message`, across all methods of
-  the API Service.
+Each type of detail payload **must** be included at most once. For example, there **must not** be more than one [`BadRequest`][BadRequest] message in the
+`details`, but there **may** be a `BadRequest` and a
+[`PreconditionFailure`][PreconditionFailure].
 
-Reasons to present an error message in `LocalizedMessage.message` that
-differs from `Status.message` include the following:
 
-- The service requires an end-user facing error message that differs
-  from the "debug message".
-- Ongoing, iterative error message improvements are expected.
+##### ErrorInfo payload
+
+The [`ErrorInfo`][ErrorInfo] message is the required way to send a machine-readable identifier. All error responses **must** include an `ErrorInfo`payload in `details`.  Variable information **should** be included in `metadata`in `ErrorInfo` and **must** be included if it appears within an error message.
+
+When introducing an error that represents a failure scenario that did not previously occur for the service, the payload **must** include `ErrorInfo` and any variables found in dynamic segments of the error message **must** be present in `ErrorInfo.metadata` (See [Dynamic variables](#dynamic-variables).
 
-When including `LocalizedMessage`, both fields, `locale` and `message`,
-**must** be populated.  If the service is to be localized, the value of
-`locale` **must** change dynamically. See,
-"[Localization](#localization)". Otherwise, `locale` **must** always
-present the service's default locale, e.g. "en-US".
+**Note:** `ErrorInfo` represents a special case. There **must** be
+exactly one `ErrorInfo`. It is required.
 
-When adding an error message via `LocalizedMessage`, `ErrorInfo`
-**must** be introduced either before or at the same time. If there are
-dynamic segments found in the text, the values of these variables
-**must** be included in `ErrorInfo.metadata`.  See, "[Dynamic
-variables](#dynamic-variables)".
+[ErrorInfo Payload][ErrorInfo]
 
-**Warning:** If `LocalizedMessage` is released without `ErrorInfo`, the
-same [concerns](#status-message-warning) regarding client misuse of
-textual error messages apply.
+Details of this object are summarized in the following fields, field
+descriptions, and examples:
 
-#### Dynamic variables
++ `reason string`
+  + A short snake_case description of why the error occurred. Error reasons are
+    unique within a particular domain of errors. The error reason **must** do the following:
+  + Be at most 63 characters and match a regular expression of
+    \`\[A-Z\]\[A-Z0-9\_\]+\[A-Z0-9\]\`, which represents UPPER_SNAKE_CASE.
+  + Be unique and consumable by machine actors for automation.
+  + Be meaningful enough for a human reader to understand what the reason 
+    refers  to.
+  + Map to one and only one cause, That is, avoid making machine actors reliant
```

Comment:
should That is be capitalized here if it's after a comma?

Reply:
Good catch. Thanks. Updated.